includes/bp-xprofile/classes/class-bp-rest-xprofile-fields-endpoint.php

```diff
@@ -843,6 +851,73 @@ public function get_xprofile_field_object( $request ) {
 		return $field;
 	}
 
+	/**
+	 * Retrieve the rendered value of a profile field.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param  string                    $value         The raw value of the field.
+	 * @param  integer|BP_XProfile_Field $profile_field The ID or the full object for the field.
+	 * @return string                                   The field value for the display context.
+	 */
+	public function get_profile_field_rendered_value( $value = '', $profile_field = null ) {
+		if ( ! $value ) {
+			return '';
+		}
+
+		$profile_field = xprofile_get_field( $profile_field );
+
+		if ( ! isset( $profile_field->id ) ) {
+			return '';
+		}
+
+		// Unserialize the BuddyPress way.
+		$value = bp_unserialize_profile_field( $value );
+
+		global $field;
+		$reset_global = $field;
+
+		// Set the $field global as the `xprofile_filter_link_profile_data` filter needs it.
+		$field = $profile_field;
+
+		/**
+		 * Apply Filters to sanitize XProfile field value.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param string $value Value for the profile field.
+		 * @param string $type  Type for the profile field.
+		 * @param int    $id    ID for the profile field.
+		 */
+		$value = apply_filters( 'bp_get_the_profile_field_value', $value, $field->type, $field->id );
+
+		// Reset the global before returning the value.
+		$field = $reset_global;
+
+		return $value;
+	}
+
+	/**
+	 * Retrieve the unserialized value of a profile field.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param  string $value The raw value of the field.
+	 * @return array         The unserialized field value.
+	 */
+	public function get_profile_field_unserialized_value( $value = '' ) {
+		if ( ! $value ) {
+			return '';
```

Comment:
The PHPDOC says it returns an array. So this should return an empty array?

```suggestion
			return array();
```

Reply:
Absolutely 😅